Giddy wireframe ui

TheCallSign opened the pull request.

TheCallSign committed:
wip(UI): basic topbar + left toggle sidebar

TheCallSign committed:
wip(TopBar): add links to demostrate routing

TheCallSign committed:
feat(Home): add homepage stub

TheCallSign committed:
wip(hash-router): add a router for in-app navigation

EternalDeiwos committed:
refactor(router): display router view inside the navigation frames

TheCallSign committed:
feat(UI): basic UI skeleton

EternalDeiwos committed:
fix(Sidebar): list items fill container

EternalDeiwos committed:
style(index): consume all space

TheCallSign committed:
feat(sidebar): add sidebar toggle

TheCallSign committed:
chore(nav): remove debug code

TheCallSign committed:
refactor(sidebar): switch from mui/Drawer to mui/List

TheCallSign committed:
refactor(sidebar): move sidebar injection to Home

TheCallSign committed:
chore(reducers): remove whitespace

Commit:
feat(UserMenu): add usermenu popover

Commit:
chore(git): update git ignore

TheCallSign committed:
refactor(nav): nested routing for /game

TheCallSign committed:
Merge branch 'giddy-wireframe-ui' of https://github.com/RhodesHvZ/hvz-hivemind into giddy-wireframe-ui

TheCallSign committed:
refactor(game): move game related components to game folder

TheCallSign committed:
feat(game): add Missions component

TheCallSign committed:
wip(game): add filler text

Commit:
feat(usermenu): add pages for user profile

TheCallSign committed:
refactor(game): redirect /#/ to /#/game

Commit:
feat(rules): add rules skeleton

Commit:
fix(HvZMap): css fix

Commit:
chore: remove debug code

Commit:
feat(HvZMap): add static map image as template

TheCallSign committed:
refactor(Sidebar): Sidebar is now a generic, modular component

Commit:
fix(rules): remove filereader component

TheCallSign committed:
refactor(router): fix redirect from /

TheCallSign committed:
fix(router): use <Switch> for routing and redirects

Commit:
feat(profile): add routing to profile pages

TheCallSign committed:
fix(css): fix spelling mistake

Commit:
refactor(app): cleanup

Commit:
Merge branch 'giddy-wireframe-ui' of github.com:RhodesHvZ/hvz-hivemind into giddy-wireframe-ui

EternalDeiwos commented:
### ToggleUserPopdownReducer

* style: Split the conditionals with comments

### SidebarReducer/Store

* refactor: mirror UserPopdown state
* Break map function into ```renderSidebarItems () {}```

### UserMenu

* The constructor serves no purpose
* See if you can make this a bit wider and dock it as close to the right hand side of the page as possible

### File Structure

Having a static folder in an already static folder is an anti-pattern. Please refactor the contents of this to be at the root app folder (alongside index.ejs)

### TopNavBar

* Use comments to separate dispatch functions
* Remove ```mapStateToProps``` and provide null to ```connect``` in its place.
* Constructor serves no purpose

### General Style

* Trailing whitespace: get yourself a plugin that displays and automatically removes trailing whitespace on file save.
* Exports should have a comment block above it:
```js
/**
 * Export
 * @ignore
 */
```
* All classes should have a @class tag in the comment block above it.
* All functions and classes should have a @description tag:
```js
/**
 * UserToolbar
 * @class
 *
 * @description
 * A Toolbar component at the top of the web app to handle any user interactions:
 * Session managament, user preferences
 */
```
* All global const values should have an @ignore tag.
* Organise imports into groups (see index.jsx)
* Be careful with tabs/spaces. Keep files "looking nice".

TheCallSign committed:
refactor(Sidebar): mirror UserPopdown state

TheCallSign committed:
refactor(UserPopdown): use consistent name

TheCallSign committed:
refactor(Sidebar): modularize map function

TheCallSign committed:
refactor(HvZMap): refactor location of image

TheCallSign committed:
refactor(TopNavBar): remove constructor, mapStateToProps

TheCallSign committed:
chore: add Export comment block

TheCallSign committed:
chore: match general coding style for the project

Clean up code to specification as stated on the pull request here:
https://github.com/RhodesHvZ/hvz-hivemind/pull/2#issuecomment-292501010

EternalDeiwos reviewed (commented):
Looks good to me. Well done sir.
